### PR TITLE
[NFC] Update coroutine intrinsics documentation and few remaining tests to opaque pointers

### PR DESCRIPTION
**Pull request by asl**

### CODE REVIEW

**Comment by asl** [comment]
I kept few no-op ptr->ptr bitcasts in order not to rename unnamed values

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM.

File: llvm/test/Transforms/Coroutines/coro-spill-promise-02.ll [hunk 1]
```diff
@@ -3,25 +3,25 @@
 
 %"class.task::promise_type" = type { [64 x i8] }
 
-declare void @consume(i32*)
+declare void @consume(ptr)
 declare void @consume2(%"class.task::promise_type"*)
```
Comment (nunoplopes) [inline llvm/test/Transforms/Coroutines/coro-spill-promise-02.ll:7]: missed this one

Comment (asl) [reply llvm/test/Transforms/Coroutines/coro-spill-promise-02.ll:7]: Nice catch, indeed!

**Review by nunoplopes — APPROVED** [review]

**Pull request merged**